xbmc/windowing/gbm/WinSystemGbm.cpp

```diff
@@ -246,3 +265,17 @@ void CWinSystemGbm::Unregister(IDispResource *resource)
     m_resources.erase(i);
   }
 }
+
+void CWinSystemGbm::OnLostDevice()
+{
+  CLog::Log(LOGDEBUG, "%s - notify display change event", __FUNCTION__);
+
+  // make sure renderer has no invalid references
+  KODI::MESSAGING::CApplicationMessenger::GetInstance().SendMsg(TMSG_RENDERER_FLUSH);
```

Comment:
this can be dropped. add m_renderManager.Flush to CVideoPlayer::OnLostDisplay

Reply:
I guess that means, I should remove it for all the windowing systems right?

Reply:
Up to you. Highly welcome but not needed in scope of this pr

Reply:
I've used the FlushRenderer that's present in VideoPlayer already.
It does m_renderManager.Flush(true).
I dropped the AppMessenger for the other windowing methods as well.

Reply:
I've used the FlushRenderer that's present in VideoPlayer already.
It does m_renderManager.Flush(true).
I dropped the AppMessenger for the other windowing methods as well.